In [2]:
#Reading train and test set
import pandas as pd

train_set = pd.read_csv("datasets/train_set.csv")
test_set = pd.read_csv("datasets/test_set.csv")

print(train_set.head(3))
print(test_set.head(3))

working_set = train_set.copy()

   age     sex    bmi    ...         charges isSmoker region_num
0   36  female  22.60    ...      18608.2620        1          1
1   18  female  39.16    ...       1633.0444        0          2
2   40  female  28.12    ...      22331.5668        1          4

[3 rows x 9 columns]
   age     sex     bmi    ...          charges isSmoker region_num
0   49  female  42.680    ...       9800.88820        0          2
1   32    male  37.335    ...       4667.60765        0          4
2   27  female  31.400    ...      34838.87300        1          1

[3 rows x 9 columns]


1. Initial set of features for X : isSmoker, age 
2. Initial set of features for Y: charges

I want predict individual insurance cost. "isSmoke" and "age" are an important factors here that could affect someone's insurance cost. As a result, I would choose "iSmoke" and "age" as my parameters for my model at first. Later, I would see if "bmi" and "region" help make the model better.

In [3]:
# Your code here
from sklearn.linear_model import LinearRegression
reg = LinearRegression()
X = working_set[["isSmoker", "age"]]
Y = working_set["charges"]
reg.fit(X, Y)

print("The score is ", reg.score(X,Y))

from sklearn.metrics import mean_squared_error
y_pred = reg.predict(X)
mse = mean_squared_error(Y, y_pred)
print("mean squared error is ",  mse)

from math import sqrt
rmse = sqrt(mse)
print("The root mean squared error is ", rmse)

The score is  0.7155021764803258
mean squared error is  41750283.75412038
The root mean squared error is  6461.445949175802


### Comments on the findings
The score here is pretty good, 0.71550. As we know, the closer to 1 the score is the better the model. However, we can definitly improve the model by adding more important parameters. The initial parameters are isSmoke and age and I intend to improve the model by adding "bmi", "region_num". 
The mean squared error is 41750283.7541, which is too much. The smaller the means squared error, the closer you are to finding the line of best fit. However, it is very difficult to get a small mean squared error.

In [4]:
from sklearn.linear_model import LinearRegression
reg = LinearRegression()
X = working_set[["isSmoker", "age", "bmi", "region_num"]]
Y = working_set["charges"]
reg.fit(X, Y)

print("The score on the train_set is ", reg.score(X,Y))

#Evaluating on the test_set
from sklearn.linear_model import LinearRegression
reg = LinearRegression()
X = test_set[["isSmoker", "age", "bmi", "region_num"]]
Y = test_set["charges"]
reg.fit(X, Y)

print("The score on test_set is ", reg.score(X,Y))

The score on the train_set is  0.7413490482280484
The score on test_set is  0.772722801913057


### Comments on the improved model

It seems like adding "bmi", "regoion_num" improves the model, and brings the score to 0.741449, which is better than the previous one. After my analysis, it is safe to say that "isSmoker", "age", "bmi", "region" affects the insurace cost very much. One of interesting findings here is that, if you smoke, your insurace costs tend to be higher. Moreover, smoking damages our body's natural defenses against the bacteria and viruses that cause pneumonia, which is my next project, classifying pneumonia from chest X-Ray images.  